lib/fs.js

```diff
@@ -537,8 +547,11 @@ function write(fd, buffer, offset, length, position, callback) {
 
   if (isArrayBufferView(buffer)) {
     callback = maybeCallback(callback || position || length || offset);
-    if (typeof offset !== 'number')
+    if (typeof offset !== 'number') {
       offset = 0;
```

Comment:
Since this PR is still semver-major I suggest to change this check (and the others that work the identical to this one) to:

```js
if (offset == null) {
  offset = 0;
} else {
  ...
}
```